app/controllers/data_access_controller.rb

```diff
@@ -300,59 +348,88 @@ def get_downloadable_access_methods(collection_id, granules, granule_params, hit
   end
 
   def get_order_access_methods(collection_id, granules, hits)
-    granule_ids = granules.map {|granule| granule['id']}
+    # Pull out the granule ids from the granule objects
+    granule_ids = granules.map { |granule| granule['id'] }
+
+    # Legacy Services contains ordering information at a granule level
     order_info = echo_client.get_order_information(granule_ids, token).body
-    orderable_count = 0 #order_info['order_information']['orderable']
 
-    defs = {}
+    # Default a count of how many granules are
+    # orderable (within order_info['order_information']['orderable'])
+    orderable_count = 0
+
+    # Creates an empty hash for the order option definitions
+    access_methods = {}
+
+    # For each granule that we requested information from we'll use
+    # the respone to create a more complete and compact list of option definitions
     Array.wrap(order_info).each do |info|
       info = info['order_information']
-      granule_id = info['catalog_item_ref']['id']
+
       orderable_count += 1 if info['orderable']
 
+      # Collect all the order option definitions guids which will define
+      # the options associated with the granules the user is requesting
       Array.wrap(info['option_definition_refs']).each do |ref|
         option_id = ref['id']
         option_name = ref['name']
 
-        defs[option_id] ||= {
+        access_methods[option_id] ||= {
           name: option_name,
-          count: 0
+          count: 0,
+          # umm_service: get_service_for_collection_with_type(collection_id, 'WEB SERVICES')
         }
-        defs[option_id][:count] += 1
+        access_methods[option_id][:count] += 1
       end
     end
 
-    defs = defs.map do |option_id, config|
-      option_def = echo_client.get_option_definition(option_id, token).body['option_definition']
-      if option_def['deprecated']
+    # Generally this will end up being a single record
+    access_methods = access_methods.map do |option_id, config|
+      Rails.logger.debug option_id
```

Comment:
Is this intended to commit?  I only ask as it is 'debug' - wondering if it snuck in there